### ISSUE TITLE
Is there any default setting for 'vocabulary' parameter in a configuration file?

### ISSUE THREAD

**Issue by entslscheia** [allenai/allennlp#3332]
I am still in the process of learning how to train models using a configuration file (json or jsonnet) instead of writing up everything manually. I have learned that for a configuration file, we need to have a top-level 'vocabulary' parameter, which is served as a global value and shared by model and iterator, however, I notice that in some situation, people don't explicitly include this parameter in their configuration files or only use fairly vague description of 'vocabulary' (e.g., the only option they specified for 'vocabulary' is 'min_count'). So if we don't tell the system how to load a vocabulary, what is allennlp gonna do? Will allennlp automatically generate a global vocabulary with corresponding namespaces based on our dataset, and then associate the automatically generated vocabulary with our model and iterator?

**Comment by matt-gardner** [comment]
Yes, that is what will happen.  We will use [default parameters](https://github.com/allenai/allennlp/blob/200239712738815a49c24f86961762b6f1636513/allennlp/data/vocabulary.py#L163-L210) for creating the vocabulary (which goes over all the instances, counts tokens in each namespace, and keeps all of them above some threshold, which is 0 by default).

**Comment by entslscheia** [comment]
For anyone who is curious about the behavior of building a default vocabulary, please look at [this](https://github.com/allenai/allennlp/blob/3dda5ac9fcf390d8b83eb855249360711707c11c/allennlp/training/trainer_pieces.py#L61-L72). Allennlp will go over all instances from datasets specified for building vocabulary (e.g., maybe only training data. By default, all datasets) to create a vocabulary.
